flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/runtime/join/SortMergeJoinOperator.java

```diff
@@ -209,17 +210,21 @@ public void processElement2(StreamRecord<BaseRow> element) throws Exception {
 		this.sorter2.write(element.getValue());
 	}
 
-	public void endInput1() throws Exception {
-		isFinished[0] = true;
-		if (isAllFinished()) {
-			doSortMergeJoin();
-		}
-	}
-
-	public void endInput2() throws Exception {
-		isFinished[1] = true;
-		if (isAllFinished()) {
-			doSortMergeJoin();
+	@Override
+	public void endInput(int inputId) throws Exception {
+		switch (inputId) {
+			case 1:
+				isFinished[0] = true;
+				if (isAllFinished()) {
+					doSortMergeJoin();
+				}
+				break;
+			case 2:
+				isFinished[1] = true;
+				if (isAllFinished()) {
+					doSortMergeJoin();
+				}
+				break;
```

Comment:
The code seems duplicated. 
Maybe you can replace with 

isFinished[inputId] = true;
if (isAllFinished()) {
    doSortMergeJoin();
}

Reply:
`isFinished[inputId - 1] = true`